casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala

```diff
@@ -451,42 +450,29 @@ class MultiParentCasperImpl[F[_]: Sync: Concurrent: ConnectionsCell: TransportLa
 
   private def reAttemptBuffer(
       dag: BlockDagRepresentation[F]
-  ): F[Unit] =
+  ): F[Unit] = {
+    import cats.instances.list._
+    type Attempt = (BlockMessage, (ValidBlockProcessing, BlockDagRepresentation[F]))
     for {
       _                    <- Span[F].mark("reattempt-buffer")
       state                <- Cell[F, CasperState].read
       dependencyFree       = state.dependencyDag.dependencyFree
       dependencyFreeBlocks = state.blockBuffer.intersect(dependencyFree).toList
-      attemptsWithDag <- dependencyFreeBlocks.foldM(
-                          (
-                            List.empty[
-                              (
-                                  BlockMessage,
-                                  (ValidBlockProcessing, BlockDagRepresentation[F])
-                              )
-                            ],
-                            dag
-                          )
-                        ) {
+      attemptsWithDag <- dependencyFreeBlocks.foldM((List.empty[Attempt], dag)) {
                           case ((attempts, updatedDag), blockHash) =>
                             for {
                               maybeBlock <- BlockStore[F].get(blockHash)
-                              b <- maybeBlock
-                                    .map(_.pure[F])
-                                    .getOrElse(
-                                      Sync[F].raiseError(
-                                        new RuntimeException(
-                                          s"Could not find a block for hash $blockHash in the blockstore. Exiting..."
-                                        )
-                                      )
+                              b <- F.fromOption(
+                                    maybeBlock,
+                                    new RuntimeException(
+                                      s"Could not find a block for hash $blockHash in the blockstore. Exiting..."
                                     )
+                                  )
                               status <- attemptAdd(b, updatedDag)
                             } yield ((b, status) :: attempts, status._2)
                         }
       (attempts, updatedDag) = attemptsWithDag
-      _ <- if (attempts.isEmpty) {
-            ().pure[F]
-          } else {
+      _ <- F.unlessA(attempts.isEmpty) {
```

Comment:
Nitpick: I find unlessA to be harder to read

Reply:
Thanks, well, IMO it's definitely better than the  `().pure[F]` ceremony (even `F.unit` would've been better) by hand. But I'm open to suggestions. What would you do instead?

Reply:
F.unit is fine. `if` statements naturally parse for me; `unlessA` takes a bit more effort.

Reply:
I think I like the fact that there is no need to specify what happens if the condition isn't met, plus you can pass it anything (it _voidifies_ any F[A]). Would you mind if I kept it?

Reply:
Ok

Reply:
:+1:

Reply:
Fixed in 142b3c680bdb5116c371380a004ad6004c81d6eb